specs/spec-1-kudos.md

```diff
@@ -1,106 +1,310 @@
+
+### Web2 Inspiration
+
+- Reddit
+- StackOverflow
+- [lemmy](https://lemmy.ml/)
+  - Open source
+
+### Web3 Inspiration
+
+- [webuidl Kudos](https://near.social/#/webuidl.near/widget/Kudos)
+  - Remove color background on each Kudo.
+  - Move upvote button to the left of each row.
+  - Use spacing that the input labels are clear which box they belong to.
+  - Input a near name instead of a url.
+  - Remove the follow/following button.
+  - On submission text inputs do not clear out and the kudo does not appear.
+  - Allow markdown in the accolade.
+  - Preview of accolade. 
+- [MetaStakingVote on NS](https://near.social/#/meta-pool-official.near/widget/MetaStakingVote)
+
+### Other potential names for this widget
 
 - High Five
 - Props
 
-## Scope and Requirements
+# Features
 
-Kudo is a Soulbound Token. **ONLY Human Verified accounts** (through `i-am-human`) can issue and receive a Kudo.
+<!-- Who is using this and what does it do? Leave "how" to the design/tech specification sections -->
 
-Sending and receiving Kudos is a form of building up a social reputation and a recognition. However there are potential risks that the system will be misused.
+## Actors / Actions
 
-### Kudo categories
+<!-- List the different groups of people that will use this widget and what each one can specifically do. If more clarity is needed you can include an actor/action matrix -->
 
-SBT standard provides a notion of classes. We will use it to define Kudo categories. Initial set of categories is following: Developer, Community, Manager, Showman, ...
+**Visitor** no credentials, not logged in.
 
-TODO: finalize the categories
+- View Kudos List
+- View Kudos Detail
+- Share Kudos (list and deep link to a Kudo detail)
 
-### Spam protection
+**User** logged in to NS.
+
+- Everything visitor can do
+- Comment on Kudos
+- React to Kudos 
+
+**Verified Human** logged in user who has credentials issued by `i-am-human`.
 
-Using proof of personhood is not enough to protect a system from abuse. One user could attach a bot to his account and send thousands of kudos to other accounts (potentially to the same human account).
-We propose to limit that activity: Any human can send maximum 14 kudos per week. The quota will reset every Sunday at midday UTC (12:00). The quota will be a parameter and we will be able to change it in the future.
+- Everything User can do
+- Create new Kudos
+- Recieve new Kudos
 
-Moreover, user should be in control of his Kudos. He will have a full right of removing an unwanted Kudo.
+**Administrator** can take additional actions not availalbe to other actors.
 
-Kudo sender will have to set an expire time. To make it more useful, the contract will require that a minimum expire time is 1month.
+- Everything Verified Human can do
+- Delete Kudos
```

Comment:
Maybe a user should be able to delete his unwanted kudos as well?

Reply:
To begin with I think it's good enough for administrators to have that right. It may be a feature that only admins can delete (in cases of behavior against COC) but it's still appropriate for a kudo to appear even if a nominee doesn't desire it. One hypothetical situation would be a nominee who wants one governance seat so they get nominated for that & want to delete their nominations for any other governance seat to improve their chances of their preferred seat, even though they would be a better fit for another seat.

Reply:
Let's describe it in the spec.